rotkehlchen/tests/unit/globaldb/test_upgrades.py

```diff
@@ -1228,6 +1240,11 @@ def test_upgrade_v12_v13(globaldb: GlobalDBHandler, messages_aggregator):
                 tuple(protocols),
             ).fetchone()[0] == expected_count
 
+        # Verify balancer cache migration: old keys gone, new keys exist with same count
+        assert cursor.execute('SELECT COUNT(*) FROM general_cache WHERE key LIKE "BALANCER_GAUGES%"').fetchone()[0] == 0  # noqa: E501
+        # `balancer_gauges_count - 1` because a malformed/unexpected balancer gauge cache entry got deleted  # noqa: E501
+        assert cursor.execute('SELECT COUNT(*) FROM general_cache WHERE key IN (?, ?, ?, ?, ?, ?, ?)', list(balancer_cache_mapping.values())).fetchone()[0] == balancer_gauges_count - 1  # noqa: E501
```

Comment:
hmm when/where/how did you put the malformed key in the DB? Can you also check the malformed key exists before?